### PR TITLE
Task/messages search/ui

### PR DESCRIPTION
**Pull request by teamagam7**

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 1]
```diff
@@ -93,10 +98,31 @@ public boolean isScrollDownFabVisible() {
...
+
   public void onScrollDownFabClicked() {
     scrollDown();
   }
 
+  public void onSearchDownFabClicked() {
+    showSearchBox();
+  }
+
+  public void onNextResultSearchCLick() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: typo in CLick

Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: and it should be click*ed*

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 2]
```diff
@@ -93,10 +98,31 @@ public boolean isScrollDownFabVisible() {
...
+    sLogger.d("Next button clicked.");
+  }
+
+  public void onPreviousResultSearchCLick() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: typo in CLick + should be Clicked

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 3]
```diff
@@ -93,10 +98,31 @@ public boolean isScrollDownFabVisible() {
...
+    sLogger.d("Previous button clicked.");
+  }
+
+  public void onEditSearchBoxResultCLick(CharSequence text) {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: typo in CLick + should be Clicked

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 4]
```diff
@@ -93,10 +98,31 @@ public boolean isScrollDownFabVisible() {
     return mIsScrollDownFabVisible;
   }
 
+  @Bindable
+  public boolean isSearchFabVisible() {
+    return mIsSearchBoxVisible;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: called fabVisible, returns boxVisible ?

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 5]
```diff
@@ -113,6 +139,19 @@ public void onLastVisibleItemPositionChanged(int position) {
     updateScrollDownFabVisibility(position);
   }
 
+  private void showSearchBox() {
+    hideSearchButton();
+  }
+
+  private void hideSearchButton() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: This doesn't necessarily hide, it just toggles the current state.
It's a little tricky to understand from this code alone that changing the BOX boolean flag is effecting the box AND the FAB.

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java [hunk 6]
```diff
@@ -113,6 +139,19 @@ public void onLastVisibleItemPositionChanged(int position) {
...
+    if (mIsSearchBoxVisible) {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: This simple boolean flip can be done with:
`bool = !bool`

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -183,6 +183,7 @@
   <string name="free_draw_title">Free draw</string>
   <string name="toast_switch_to_pan_text">Switch mode to Pan</string>
   <string name="default_timeplay_date_string" translatable="false">-</string>
+  <string name="search">Search</string>
```
Comment (BarAgam) [inline app/src/main/res/values/strings.xml]: Don't forget to add an Hebrew string as-well....

**Review by BarAgam — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 4 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 1 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 2 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 3 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 5 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java (hunk 6 above)
Comment (teamagam7) [reply app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/MessagesViewModel.java]: done

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (teamagam7) [reply app/src/main/res/values/strings.xml]: done

**Pull request merged**